docs/development.md

```diff
@@ -91,3 +91,101 @@ The `cbindgen` xtask generates multiple header files for four different modules:
 4. The types used by the C++ interpreter API, written to `slint_interpreter_internal.h`.
 
 Typically the input to `cbindgen` is within `ffi` sub-modules in the corresponding input crates to `cbindgen`. These `ffi` modules are gated with `#[cfg(feature = "ffi")]`.
+
+## Commit History & Code Reviews
+
+Linear history is preferred over merge commits. Long lived features can live in feature branches and those can be integrated
+with merge commits, of course.
+
+As a consequence, we typically integrate pull requests as "rebase and merge" or "squash and merge".
+
+During code review, consider adding small commits on top to make it easier for the reviewer and contributor to track feedback and
+how the feedback was incorporated - rebase can tend to make it harder. It's perfectly fine to then squash these commits when the
+review phase is complete and approval is given.
+
+Example:
+
+A PR consists of three commits:
+
+1. Add new widget
+2. Add documentation for new widget
+3. Change example to use new widget
+
+In the review phase, the reviewer suggests to make changes to the widget implementation and the documentation. Consider pushing
+these as follow-up fixes:
+
+1. Add new widget
+2. Add documentation for new widget
+3. Change example to use new widget
+4. Fix an issue in widget found during review
+5. Fix a typo in the documentation
+
+Finally, the PR is approved. As contributor, in your local branch, feel free to merge 4. into 1. and 5. into 2.:
+
+(commits are real, sha1s are just examples)
+
+Find the base commit:
+```
+$ git merge-base HEAD origin/master
+607bdbfcf80a879e09d2d92b0d3e195d426fb481
+$ merge_base=607bdbfcf80a879e09d2d92b0d3e195d426fb481
+```
+
+Start an interactive rebase that starts at the base commit:
+
+```
+$ git rebase -i $merge_base
```

Comment:
I usually do  git rebase -i origin/master
I think there is no need to introduce the merge base there.  

But if you think we should recommand to use the merge-base, the docs would probably be to say   `git rebase -i 607bdbfcf80a879e09d2d92b0d3e195d426fb481`  because there is not reason to introduce a variable, pratcically.

Reply:
I'm torn. Initially I had the sha1 here, but I'm worried that this might be taken as copy & paste, and we really don't want anyone to accidentally rebase on `607bdbfcf80a879e09d2d92b0d3e195d426fb481` (literally).

The issue I had with `origin/master` is that this make result in additional stops due to conflict resolution.

But I guess it's the cleaner approach overall, even if it means more work (two steps in one). I'll try to rephrase this.